### ISSUE TITLE
FAC Debugging Django w/ "new" s3

### ISSUE THREAD

**Issue by asteel-gsa** [GSA-TTS/FAC#2657]

**Comment by tadhg-ohiggins** [comment]
The following works in both `dev`and `staging`:

```sh
python
```

```python
import boto3
from os import environ
import json

vc = environ.get("VCAP_SERVICES")
vcj = json.loads(vc)
vcj.keys()
vcj["s3"]
len(vcj["s3"])
bucket = next(b for b in vcj["s3"] if b["instance_name"] == "fac-private-s3")

client = boto3.client('s3', region_name='us-gov-west-1', endpoint_url=f"https://{bucket['credentials']['endpoint']}", aws_access_key_id=bucket['credentials']['access_key_id'], aws_secret_access_key=bucket['credentials']['secret_access_key'])

res = client.put_object(Bucket=bucket['credentials']["bucket"], Key='datadata', Body=b'datadata')
res2 = client.list_objects(Bucket=bucket['credentials']["bucket"])
res2["Contents"][0].keys()
[x for x in res2["Contents"] if x["Key"] == "datadata"]
```
Given the above, this is where we currently think we are:

*   Using `boto3` directly from either Python or a Django shell in both `dev` and `staging` allows us to successfully upload a file to our S3 buckets.
*   If we use our app’s functionality in Django shell, we can upload a file in `staging` but not `dev`.
*   If we use our app’s `report_submission.S3PrivateStorage`, which is what our application uses for file upload, in the Django shell we can upload in `staging` but not `dev`.
*   We think it’s possible that Django storages, which we use in the app, is trying to set the ACL at the same time that it uploads files, and that an April change to AWS bucket permissions may mean that this would have worked when `staging` and `production` S3 buckets were created, but not when `preview` or the recent `dev` S3 buckets were created S3 buckets were created. See [this discussion](https://gsa-tts.slack.com/archives/C09CR1Q9Z/p1698437115017019?thread_ts=1698432989.403159&cid=C09CR1Q9Z). (We think that our `staging` and `production` S3 buckets were created pre-April, and our `preview` S3 bucket post-April, and our `dev` S3 bucket was created today.)

**Comment by asteel-gsa** [comment]
@timoballard was able to validate that the following:
```python
import boto3
from django.conf import settings
settings.AWS_DEFAULT_ACL = "private" # This being the core line
session = boto3.session.Session(aws_access_key_id=settings.AWS_PRIVATE_ACCESS_KEY_ID, aws_secret_access_key=settings.AWS_PRIVATE_SECRET_ACCESS_KEY, region_name=settings.AWS_S3_PRIVATE_REGION_NAME)
s3 = session.resource("s3", endpoint_url=settings.AWS_S3_ENDPOINT_URL)  # No, this does not have _PRIVATE in the variable name.
bucketname = settings.AWS_PRIVATE_STORAGE_BUCKET_NAME
bucket = s3.Bucket(bucketname)
from os import environ  # Paranoia: just make sure we're on dev
environ
environ["no_proxy"]

from audit.models import SingleAuditReportFile
from audit.models import SingleAuditChecklist
from django.core.files.base import File
somefile = open('/app/manage.py', "rb")
djangofile = File(file=somefile)
print(settings.AWS_PRIVATE_ACCESS_KEY_ID)  # Capture this for comparison with creds from local step below.
sac = SingleAuditChecklist.objects.filter(report_id="2023-09-GSAFAC-0000000001")[0]
sar = SingleAuditReportFile.objects.create(file=djangofile,filename='whatever',sac=sac,component_page_numbers=[1,1,1,1,1,1,1])
```

was able to submit to the s3 bucket

**Comment by asteel-gsa** [comment]
This has been fixed with #2663
